internal/common/baseclient/basejudge.go

```diff
@@ -0,0 +1,87 @@
+package baseclient
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/pkg/errors"
+)
+
+type BaseJudge struct {
+}
+
+// PayPresident pays the President a salary.
+func (j *BaseJudge) PayPresident(presidentSalary shared.Resources) shared.Resources {
+	// TODO Implement opinion based salary payment.
+	return presidentSalary
+}
+
+// inspectHistoryInternal is the base implementation of InspectHistory.
+func (j *BaseJudge) InspectHistory() (map[shared.ClientID]roles.EvaluationReturn, error) {
+	outputMap := map[shared.ClientID]roles.EvaluationReturn{}
+	for _, v := range gamestate.TurnHistory {
+		variablePairs := v.Pairs
+		clientID := v.ClientID
+		var rulesAffected []string
+		for _, v2 := range variablePairs {
+			valuesToBeAdded, err := PickUpRulesByVariable(v2.VariableName, rules.RulesInPlay)
+			if err == nil {
+				rulesAffected = append(rulesAffected, valuesToBeAdded...)
+			}
+			err = rules.UpdateVariable(v2.VariableName, v2)
+			if err != nil {
+				return map[shared.ClientID]roles.EvaluationReturn{}, err
+			}
+		}
+		if _, ok := outputMap[clientID]; !ok {
+			tempTemp := roles.EvaluationReturn{
+				Rules:       []rules.RuleMatrix{},
+				Evaluations: []bool{},
+			}
+			outputMap[clientID] = tempTemp
+		}
+		tempReturn := outputMap[clientID]
+		for _, v3 := range rulesAffected {
+			evaluation, _ := rules.BasicBooleanRuleEvaluator(v3)
+			tempReturn.Rules = append(tempReturn.Rules, rules.RulesInPlay[v3])
+			tempReturn.Evaluations = append(tempReturn.Evaluations, evaluation)
+		}
+	}
+	return outputMap, nil
+}
+
+// DeclareSpeakerPerformance checks how well the speaker did their job.
+func (j *BaseJudge) DeclareSpeakerPerformance(inspectBallot bool, conductedRole bool) (result bool, didRole bool) {
+	return inspectBallot, conductedRole
+}
```

Comment:
Short answer - yes, but not at the moment.

DeclareSpeakerPerformance broadcasts to all the islands the judge's 'opinion' of how well the speaker has done its job. This function is overridable by individual islands and will eventually be implemented by each team. At each team's discretion, this performance message will be based on the Judge's own opinion of the Speaker as well as the quantitative metrics of the Speaker's performance, i.e. inspectBallot and conductedRole which are inputs to this function. In MVP there was no opinion formation, therefore this performance statement is trivial and serves merely as a placeholder for post-MVP work. 

I will add in a TODO to make this clear.

Reply:
Okay that makes sense. Thanks! I think in this case it will be a COMPULSORY instead of a TODO since you want it to be overridden by cleints. [Check out internal/common/baseclient/iifo.go to see an example of function description and the COMPULSORY/OPTIONAL comments at work]